tmp/keycloak.py

```diff
@@ -0,0 +1,102 @@
+import requests
+import click
+import json
+import sys
+import base64
+
+TOKEN_ENDPOINT = "realms/redhat-external/protocol/openid-connect/token"
+USER_ENDPOINT = "admin/realms/redhat-external/users"
+SIGNUP_ENDPOINT = "api/v1/signup"
+
+@click.command(context_settings=dict(help_option_names=["-h", "--help"]))
+@click.argument("kc", type=str, required=True)
+@click.argument("admin", type=str, required=True)
+@click.argument("password", type=str, required=True)
+@click.argument("user", type=str, required=True)
+@click.argument("secret", type=str, required=True)
+@click.argument("register", type=str, required=True)
+def main(kc, admin, password, user, secret, register):
+    tokenUrl = f"{kc}{TOKEN_ENDPOINT}"
+    userUrl = f"{kc}{USER_ENDPOINT}"
+    registerUrl = f"{register}{SIGNUP_ENDPOINT}"
+    params = {
+        "grant_type": "password",
+        "username": admin,
+        "password": password,
+        "scope": "openid",
+        "client_id": "admin-cli",
+    }
+    headers = {
+        "Content-Type": "application/x-www-form-urlencoded",
+    }
+    
+    r = requests.post(tokenUrl, data=params, headers=headers)
+    access_token = r.json()["access_token"]
+
+    headers["Content-Type"] = "application/json"
+    headers["Authorization"] = f"Bearer {access_token}"
+
+    decodedUser = secret_to_string(user)
+    decodedPassword = secret_to_string(secret)
+
+    user = {
+        "enabled": "true",
+        "username": decodedUser,
+        "firstName": "Ephemeral",
+        "lastName": "Environment",
+        "email":  f"{decodedUser}@redhat.com",
+        "attributes": {
+            "firstName": "Ephemeral",
+            "lastName": "Environment",
+            "AccountID": "12345",
+            "AccountNumber": 12345,
+            "OrgID": "12345",
+            "IsInternal": "false",
+            "IsOrgAdmin": "true",
+            "IsActive": "true",
+            "Entitlements": '"insights": {"is_entitled": "true", "is_trial": "false"}'
```

Comment:
Do we need to send `insights` entitlements? I think it should be enough to just send empty object.

Reply:
I talked with Pete, and supporting this is going to be a bit of an effort to update hac-proxy. The user we're adding is already registered and we'd like to get this merged for e2e support.